### DISCUSSION TITLE
Magick++ documentation for "Quantum" wrong

### DISCUSSION THREAD

**Discussion by M-Reimer** [ImageMagick/ImageMagick#6452] [Help]
### ImageMagick version

GIT version from 2022-07-20

### Operating system

Linux

### Operating system, version and so on

Arch

### Description

I want to read the red, green, blue and opacity channel from all pixels in an image.

To do so, I call "getConstPixels" which returns a "Quantum".
But the description, hosted here, seems to be wrong:
https://imagemagick.org/Magick++/Quantum.html

What's wrong here? How can I actually get the four channels I want to get? Seems like "Quantum" is not a member of "Magick++" at all as my compiler says:

    error: request for member ‘opacity’ in ‘* pix’, which is of non-class type ‘const MagickCore::Quantum’ {aka ‘const float’}

### Steps to Reproduce

Try to access any channel of a "Magick::Quantum" as described in the documentation. Does not work as Magick++ seems to be incomplete here.

**Comment by snibgo** [comment]
I think that page was correct for integer IM 6, and not for IMv7, nor for HDRI.

I suggest you follow the example programs supplied. And, of course, the internal code of ImageMagick.

**Comment by M-Reimer** [comment]
I've tried but did not find something that helps. There does not seem to be an example that just reads RGBA and most internal code uses MagickCore and not Magick++.

**Comment by M-Reimer** [comment]
I've kept trying but "GetPixelRed" is not defined in "Magick::" namespace and if I use "MagickCore::GetPixelRed", then I get:

    error: cannot convert ‘Magick::Image’ to ‘const MagickCore::Image*’ {aka ‘const MagickCore::_Image*’}

Seems like Magick++ is currently broken if it gets to reading pixels.

Edit: Is it possible to get a "Magick::Color" for each pixel while running through them? All I'm searching is a nice object oriented way to get pixels and their color...

**Comment by unknown321** [comment]
```c
Magick::Color green = Magick::Color("green");

for (ssize_t row = 0; row < source.rows(); ++row)
    for (ssize_t column = 0; column < source.columns(); ++column)
        source.pixelColor(column, row, green);
```
